### PR TITLE
1.0.0 release

### LINKED ISSUE #1645: Version 1.0 release strategy
We have a loosely communicated intent to ship python-tuf 1.0 with only the new/refactored code (metadata API + ngclient + ???). We should define what that release looks like, including accounting for:

- Communicating deprecation expectations with regards the legacy code
  - [ADR 2](https://github.com/theupdateframework/python-tuf/blob/develop/docs/adr/0002-pre-1-0-deprecation-strategy.md) indicates legacy code will be supported on a best-effort basis, but this has not been communicated to users of python-tuf
  - > "Support for older releases: Bugs reported with tuf versions prior to 1.0.0 will likely not be addressed directly by tuf’s maintainers. Pull Requests to fix bugs in the last release prior to 1.0.0 will be considered, and merged (subject to normal review processes). Note that there may be delays due to the lack of developer resources for reviewing such pull requests."
- What the feature complete state looks like for a python-tuf 1.0 release? Must python-tuf 1.0 provide feature parity with the legacy code?

### LINKED ISSUE #1845: Write 1.0 changelog / release notes
In preparation for release, start working on changelog

### PR DESCRIPTION
**Pull request by jku**
This contains:
* removal of an unused constant (we could drop the commit as well, it's not too bad)
* PR #1869 as is
* PR #1851 just  with version number changes added in the same commit

I think it makes sense to do the last two here and close the other PRs without merging -- the constant removal could be dropped as I mentioned.

I'm running release tests now, but review is welcome

### CODE REVIEW

**Comment by jku** [comment]
force pushed to add my signed-off-bys to Lukas commits

File: setup.cfg [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [metadata]
 name = tuf
-version = 0.20.0
+version = 1.0.0
```
Comment (lukpueh) [inline setup.cfg:3]: ❤️

**Comment by lukpueh** [comment]
Nit: Mind squashing my two doc commits into one?

**Comment by jku** [comment]
force-pushed: no changes, but squashed a commit to one.

**Review by lukpueh — COMMENTED** [review]
Just to make sure, are the following PRs not mentioned in the changelog on purpose, due to insignificance?

- #1765
- #1794
- #1805
- #1814
- #1821
- #1823
- #1824

They were all merged after our last release 0.20.0
```
git log --oneline --merges v0.20.0..HEAD | egrep -o --color=never "Merge pull request #[0-9]+" | cut -d" " -f4
```

**Comment by jku** [comment]
yes, looks correct -- I don't object to including them, they just seemed not relevant

**Review by lukpueh — APPROVED** [review]
🎉  Thanks for preparing the release, @jku!

Here's what I did to review: I checked that
- the already approved [#1869](https://github.com/theupdateframework/python-tuf/pull/1869) is fully included in this PR,
- all significant PRs are included in the changelog (see [comment above](https://github.com/theupdateframework/python-tuf/pull/1873#pullrequestreview-887337692)),
- all changelog items list the corresponding PRs,
-  and I proof-read the changelog

**Comment by jku** [comment]
I'll have another quick look and will then get this done...

**Pull request merged**